crates/node/src/shims/abcipp_shim.rs

```diff
@@ -114,43 +113,89 @@ impl AbcippShim {
+                                        // for finalize block (tx results)
+                                        let process_req = super::abcipp_shim_types::shim::request::CheckProcessProposal::from(begin_block_request.clone()).cast_to_tendermint_req(self.delivered_txs.clone());
 
-                    let processing_results = self.service.process_txs(
-                        &self.delivered_txs,
-                        block_time,
-                        &block_proposer,
-                    );
-                    let mut txs = Vec::with_capacity(self.delivered_txs.len());
-                    let mut delivered = vec![];
-                    std::mem::swap(&mut self.delivered_txs, &mut delivered);
-                    for (result, tx) in processing_results
-                        .into_iter()
-                        .zip(delivered.into_iter())
-                    {
-                        txs.push(ProcessedTx { tx, result });
-                    }
-                    let mut end_block_request: FinalizeBlock =
-                        begin_block_request.into();
-                    end_block_request.txs = txs;
-                    self.service
+                                        let (process_resp, res) =
+                                            self.service.process_proposal(
+                                                process_req.into(),
+                                            );
+                                        // Cache the result
+                                        self.service
+                                            .state
+                                            .in_mem_mut()
+                                            .process_proposal_cache
+                                            .insert(
+                                                block_hash.to_owned(),
+                                                (
+                                                    process_resp,
+                                                    res.clone()
+                                                        .into_iter()
+                                                        .map(|res| res.into())
+                                                        .collect(),
+                                                ),
+                                            );
+
+                                        (process_resp, res)
+                                    }
+                                }
+                            }
+                            Err(_) => {
+                                // Need to run process proposal to extract the
+                                // data we need
+                                // for finalize block (tx results)
+                                let process_req = super::abcipp_shim_types::shim::request::CheckProcessProposal::from(begin_block_request.clone()).cast_to_tendermint_req(self.delivered_txs.clone());
```

Comment:
again, can we try to fix formatting here?

Reply:
Done in 33a4049e0b0b87b6fd68b560d3d2e3a43663ea8e